src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -88,7 +88,7 @@ internal static partial class ICodeDefinitionFactoryExtensions
 
             yield return CodeGenerationSymbolFactory.CreateConstructorSymbol(
                 attributes: default(ImmutableArray<AttributeData>),
-                accessibility: Accessibility.Public,
+                accessibility: containingTypeOpt.IsAbstractClass() ? Accessibility.Protected : Accessibility.Public,
```

Comment:
It feels like CreateBaseDelegatingConstructor (method above this one) would also need to be updated.